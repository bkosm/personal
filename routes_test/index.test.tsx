import { browserTest, assertSnapshot } from "../test.deps.ts";

browserTest("index page", async (b, t) => {
  const page = await b.newPage();

  await page.goto("http://localhost:8000/", {
    waitUntil: "networkidle2",
  });

  await t.step("text loads", async () => {
    const pElem = await page.waitForSelector("div > p");
    const pText = await pElem?.evaluate((el) => el.textContent);

    assertSnapshot(t, pText);
  });

  await t.step("visual requirements are met", async () => {
    // await page.screenshot({
    //   fullPage: true,
    //   path: `${SNAPSHOT_DIR}/index.png`,
    // });
  });
});
