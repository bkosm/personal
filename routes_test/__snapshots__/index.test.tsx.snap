export const snapshot = {};

snapshot[`index page 1`] = `"Welcome to \`fresh\`. Try updating this message in the ./routes/index.tsx file, and refresh."`;
